pkg/sdk/connections_def.go

```diff
@@ -0,0 +1,121 @@
+
+import (
+	g "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/poc/generator"
+)
+
+var ConnectionDef = g.NewInterface(
+	"Connections",
+	"Connection",
+	g.KindOfT[AccountObjectIdentifier](),
+).CreateOperation(
+	"https://docs.snowflake.com/en/sql-reference/sql/create-connection",
+	g.NewQueryStruct("CreateConnection").
+		Create().
+		SQL("CONNECTION").
+		IfNotExists().
+		Name().
+		OptionalComment().
+		WithValidation(g.ValidIdentifier, "name"),
+).CustomOperation(
+	"CreateReplicated",
+	"https://docs.snowflake.com/en/sql-reference/sql/create-connection",
+	g.NewQueryStruct("CreateReplicated").
+		Create().
+		SQL("CONNECTION").
+		IfNotExists().
+		Name().
+		Identifier("ReplicaOf", g.KindOfT[ExternalObjectIdentifier](), g.IdentifierOptions().Required().SQL("AS REPLICA OF")).
+		OptionalComment().
+		WithValidation(g.ValidIdentifier, "name").
+		WithValidation(g.ValidIdentifier, "ReplicaOf"),
+).CustomOperation(
+	"AlterFailover",
+	"https://docs.snowflake.com/en/sql-reference/sql/alter-connection",
+	g.NewQueryStruct("AlterFailover").
+		Alter().
+		SQL("CONNECTION").
+		Name().
+		OptionalQueryStructField(
+			"EnableConnectionFailover",
+			g.NewQueryStruct("EnableConnectionFailover").
+				List("ToAccounts", "AccountIdentifier", g.ListOptions().NoParentheses()).
+				OptionalSQL("IGNORE EDITION CHECK"),
+			g.KeywordOptions().SQL("ENABLE FAILOVER TO ACCOUNTS"),
+		).
+		OptionalQueryStructField(
+			"DisableConnectionFailover",
+			g.NewQueryStruct("DisableConnectionFailover").
+				OptionalQueryStructField(
+					"ToAccounts",
+					g.NewQueryStruct("ToAccounts").
+						List("Accounts", "AccountIdentifier", g.ListOptions().NoParentheses()),
+					g.KeywordOptions().SQL("TO ACCOUNTS"),
+				),
+			g.KeywordOptions().SQL("DISABLE FAILOVER"),
+		).
+		OptionalSQL("PRIMARY").
+		WithValidation(g.ExactlyOneValueSet, "EnableConnectionFailover", "DisableConnectionFailover", "Primary"),
+).AlterOperation(
+	"https://docs.snowflake.com/en/sql-reference/sql/alter-connection",
+	g.NewQueryStruct("Alter").
+		Alter().
+		SQL("CONNECTION").
+		IfExists().
+		Name().
+		OptionalQueryStructField(
+			"Set",
+			g.NewQueryStruct("Set").
+				OptionalComment().
+				WithValidation(g.AtLeastOneValueSet, "Comment"),
+			g.KeywordOptions().SQL("SET"),
+		).
+		OptionalQueryStructField(
+			"Unset",
+			g.NewQueryStruct("Unset").
+				OptionalSQL("COMMENT").
+				WithValidation(g.AtLeastOneValueSet, "Comment"),
+			g.KeywordOptions().SQL("UNSET"),
+		).
+		WithValidation(g.ExactlyOneValueSet, "Set", "Unset"),
+).DropOperation(
+	"https://docs.snowflake.com/en/sql-reference/sql/drop-connection",
+	g.NewQueryStruct("DropConnection").
+		Drop().
+		SQL("CONNECTION").
+		IfExists().
+		Name().
+		WithValidation(g.ValidIdentifier, "name"),
+).ShowOperation(
+	"https://docs.snowflake.com/en/sql-reference/sql/show-connections",
+	g.DbStruct("connectionRow").
+		Field("snowflake_region", "string").
+		Field("created_on", "time.Time").
```

Comment:
What about `region_group`? In [docs](https://docs.snowflake.com/en/sql-reference/sql/show-connections#output) is stated that it's only displayed for organizations that span multiple region groups, but we still should have it.

Reply:
good point, it should be added here